tools/coqdep/lib/fl.ml

```diff
@@ -36,27 +36,33 @@ module Error = struct
     | _ -> None
 end
 
-let rec find_META meta_files package =
-  match meta_files with
-  | [] ->
-    (try Some (Findlib.package_meta_file package)
-     with _ -> None)
-  | m :: ms ->
-    if String.equal (Filename.extension m) ("." ^ package)
-    then Some m
-    else find_META ms package
-
+(* Platform build is doing something weird with META, hence we parse
+   when finding, but at some point we should find, then parse. *)
 let parse_META meta_file package =
   try
     let ic = open_in meta_file in
     let m = Fl_metascanner.parse ic in
     close_in ic;
-    m
+    Some m
   with
+  (* This should not be necessary, but there's a problem in the platform build *)
+  | Sys_error _msg -> None
   (* findlib >= 1.9.3 uses its own Error exception, so we can't catch
      it without bumping our version requirements. TODO pass the message on once we bump. *)
   | _ -> Error.cannot_parse_meta_file package ""
 
+let rec find_META meta_files package =
```

Comment:
maybe the correct name is the find_parseable_META

Reply:
Yes, amended. Something weird is going on in Windows, when we change this to find meta + parse the thing doesn't work. Seems very fishy and will likely be a blocker to remove the legacy layer.

Unfortunately I am at a loss on how to debug our Windows build, the platform job doesn't seem to store the logs @MSoegtropIMC  correct?